server/rest/src/main/java/org/infinispan/rest/resources/StaticFileResource.java

```diff
@@ -98,12 +98,9 @@ private File resolve(String resource) {
       responseBuilder.addProcessedDate(now);
       responseBuilder.header("Expires", DateUtils.toRFC1123(now.getTime() + TimeUnit.SECONDS.toMillis(CACHE_MAX_AGE_SECONDS)));
 
-      String mediaType = APPLICATION_OCTET_STREAM_TYPE;
-      try {
-         String probed = Files.probeContentType(file.toPath());
-         if (probed != null) mediaType = probed;
-      } catch (IOException ignored) {
-      }
+      String resolved = MediaTypeResolver.getMediaType(file.getName());
+      String mediaType = resolved != null ? resolved : APPLICATION_OCTET_STREAM_TYPE;
```

Comment:
can't his be the default coming from the resolver instead of null ?

Reply:
I was in doubt if the lookup itself should enforce the default as ```application/octet-stream```. Returning ```null``` communicates better the fact that the supplied map does not exist, and the consumer of the API is free to choose whatever it wants....